lib/improver/blending/weights.py

```diff
@@ -238,12 +242,575 @@ def process_coord(cube, coordinate, coord_exp_vals=None,
+        for example, the weights vary along the forecast_period coordinate.
+
+        Args:
+            weighting_coord_name (str):
+                Name of the coordinate along which the weights will be
+                calculated. For example, if the intention is to provide
+                weights along the forecast_period coordinate within the
+                configuration dictionary, which can be linearly interpolated
+                to create all required weights, then this argument would be
+                "forecast_period".
+                This must be included within the configuration dictionary.
+            config_coord_name (str):
+                Name of the coordinate used for configuration. For example,
+                if the intention is to create weights for different models,
+                then a model_configuration may appropriate.
+
+        """
+        self.weighting_coord_name = weighting_coord_name
+        self.config_coord_name = config_coord_name
+
+    def __repr__(self):
+        """Represent the configured plugin instance as a string."""
+        msg = ("<ChooseWeightsLinearFromCube weighting_coord_name = {}, "
+               "config_coord_name = {}>".format(
+                   self.weighting_coord_name, self.config_coord_name))
+        return msg
+
+    def _check_weights_cubes(self, cube, weights_cubes):
+        """Check that the number of weights cubes provided matches the
+        number of points along the self.config_coord_name dimension within
+        the input cube.
+
+        Args:
+            cube (iris.cube.Cube):
+                Cube containing the coordinate information that will be used
+                for checked.
+            weights_cube (iris.cube.CubeList):
+                CubeList where the number of cubes is expected the match
+                the number of points along the self.config_coord_name
+                dimension within the input cube. For example, if the
+                model_configuration dimension within the input cube has a
+                length of 2, then it is expected that there will be 2 cubes
+                within the weights_cubes CubeList.
+
+        Raises:
+            ValueError: If the number of cubes in weights_cubes does not
+                match the number of points along the self.config_coord_name
+                dimension.
+        """
+        if (len(cube.coord(self.config_coord_name).points) !=
+                len(weights_cubes)):
+            msg = ("The coordinate used to configure the weights needs to "
+                   "have the same length as the number of weights cubes. "
+                   "\n{} is {} != number of weights cubes is {}".format(
+                       self.config_coord_name,
+                       len(cube.coord(self.config_coord_name).points),
+                       len(weights_cubes)))
+            raise ValueError(msg)
+
+    def _arrange_interpolation_inputs(self, cube, weights_cube):
```

Comment:
Suggest `_generate_inputs_from_cube` or similar (see comment on same function in `ChooseWeightsLinearFromDict` plugin).